rfcs/text/responsiveGrid.md

```diff
@@ -0,0 +1,152 @@
+
+# Table of contents
+
+- [Summary](#summary)
+- [Basic example](#basic-example)
+- [Motivation](#motivation)
+- [Detailed design](#detailed-design)
+  - [CSS Grid Layout](#css-grid-layout)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [CSS Flexbox](#css-flexbox)
+  - [CSS Framework (Bootstrap or Foundation)](#css-framework-bootstrap-or-foundation)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+
+# Summary
+
+A responsive grid system is a layout component that provides guidance on how components should be positioned within a user interface, adapting to screen size and orientation. We, the Carbon team, would like to implement the grid system designed by Sage UI designers.
+
+# Basic example
+
+http://designsystem.sage.com/product/invoice/
+
+# Motivation
+
+The grid layout is intended to solve a number of problems that are often experienced when developing complex web applications:
+1. CSS floats are not well suited to application layout.
+2. Layouts that use a combination of tables, JavaScript and floated elements are often brittle and can behave unexpectedly.
+3. Fixed layouts that behave predictably cannot take advantage of the entire screen real estate.
+
+A well designed grid layout will be:
+1. Consistent, which helps users navigate and understand where information will be found.
+2. Responsive, which gives users a predictable experience across multiple devices with different screen sizes.
+3. Adaptable, enabling new design themes to be introduced more quickly as designs evolve over time.
+
+# Detailed design
+
+By dividing the available space into predictably-sized modules (both vertically and horizontally), complex layouts that don't have fixed dimensions can be controlled and components can be positioned allowing a clean separation of content and style. Once the underlying structure of a design is mapped, breakpoints can be used to control how each component responds at given widths.
+
+For maximum flexibility, the Design System responsive grid uses 12 equal-width columns. 12 is divisible into 12, 6, 4, 3, 2 or 1 equal-width columns, giving maximum possible flexibility for a layout.
+
+For full details of the proposal, read the [Sage Design System Grid](http://designsystem.sage.com/foundations/grid/).
+
+
+
+To avoid introducing a breaking change, and to give developers the freedom to choose whether or not to use the component, a new `Grid` Component will be developed, that can be used at the same level as the `<AppWrapper />`, as a grid wrapper for components. 
+The `<GridLayout />` will supply props to define breakpoints and number of columns, and will be styled using CSS Grid Layout.
+
+Code example:
+```jsx
+import AppWrapper from 'carbon-react/lib/components/app-wrapper';
+  import { GridLayout, GridRow, GridColumn }  from 'carbon-react/lib/components/grid';
+  
+  const Index = () => (
+  <AppWrapper>
+    <GridLayout 
+      breakpoints={[100, 200, 300, 400, 500]}
```

Comment:
Can we expand a bit more on what this would represent?

The design breakpoints call for:

<table>
    <tr>
      <th>Breakpoints (measured in dpª)</th>
      <th>Device Type</th>
      <th>Margins</th>
      <th>Gutters</th>
    </tr>
    <tr>
      <td>Extra Small (320 to 599px)</td>
      <td>Smart Phones</td>
      <td>16px</td>
      <td>16px</td>
    </tr>
    <tr>
      <td>Small (600 to 959px)</td>
      <td>Portrait Tablets</td>
      <td>24px</td>
      <td>16px</td>
    </tr>
    <tr>
      <td>Medium (960 to 1259px)</td>
      <td>Landscape Tablets &amp; Low-Res Laptops</td>
      <td>32px</td>
      <td>24px</td>
    </tr>
    <tr>
      <td>Large (1260 to 1920px)</td>
      <td>High-Res Laptops &amp; Monitors</td>
      <td>40px</td>
      <td>24px</td>
    </tr>
    <tr>
      <td>Extra Large (1921px and above)</td>
      <td>Ultra High-Res Monitors</td>
      <td>40px</td>
      <td>24px</td>
    </tr>
  </table>

ª Density-independent Pixels (dp) - a unit based on the physical density of the screen, so it's universal across devices with different densities, like Apple Retina displays, for example.

Reply:
Would also be interested to see thoughts on how we set the margins/gutters too

Reply:
The way breakpoints, order and units are set has been revised. Gutters will be set automatically.

Reply:
The way breakpoints, order and units are set has been revised. Gutters will be set automatically.